connector/connect/server/src/main/scala/org/apache/spark/sql/connect/artifact/SparkConnectArtifactManager.scala

```diff
@@ -154,6 +154,8 @@ class SparkConnectArtifactManager private[connect] {
         val canonicalUri =
           fragment.map(UriBuilder.fromUri(target.toUri).fragment).getOrElse(target.toUri)
         sessionHolder.session.sparkContext.addArchive(canonicalUri.toString)
+      } else if (remoteRelativePath.startsWith(s"files${File.separator}")) {
+        sessionHolder.session.sparkContext.addFile(target.toString)
```

Comment:
I believe jars have the same problem, and I believe we could share the same fix.

Reply:
My only problem here is that we need to design this for the python side. In practice artifacts for a session are exposed in a session specific location. How would a python user interact with these files? Through `org.apache.spark.SparkFiles`?

Reply:
For regular files and archives, I don't intend to expose `org.apache.spark.SparkFiles` for now.
Since the files are archives are always stored at the current working directory of executors in production, I was simply thinking about creating a session dedicated directory, and change the current working directory to that.

Meaning that the end users would continue accessing to their file with `./myfile.txt` or `./myarchive`.

Reply:
(`SparkFiles` is being used in the test case here but that's a sort of hack to make sure of cleaning up, etc.)

Reply:
Just to be extra clear, each Spark Connect session will have a dedicated directory on worker nodes.

Reply:
I will make a PR right away after this.

Reply:
Offline discussed with Herman. Will merge this first, and make a PR to address it.